x-pack/test/functional/page_objects/gis_page.js

```diff
@@ -194,20 +194,20 @@ export function GisPageProvider({ getService, getPageObjects }) {
       return hits;
     }
 
-    async gotoMapListingPage() {
+    async gotoMapListingPage(options = { isOnUnsavedMap: false }) {
       log.debug('gotoMapListingPage');
       const onPage = await this.onMapListingPage();
       if (!onPage) {
         await retry.try(async () => {
-          await PageObjects.common.navigateToUrl('maps', '/', { basePath: this.basePath });
+          await PageObjects.common.navigateToUrl('maps', '/', { basePath: this.basePath, shouldAcceptAlert: options.isOnUnsavedMap });
```

Comment:
Could we just have this method click the Maps icon in the nav bar, and then handle the alert right here and get rid of navigateToUrlAndHandleAlert ?

The Maps icon in the nav bar doesn't have a unique data-test-subj but it could get one, or use something like;
`[data-test-sub="navDraweAppsMenuLink"] [aria-label="Maps"]`